docs/spark_expressions_support.md

```diff
@@ -0,0 +1,477 @@
+ - [ ] approx_percentile
+ - [ ] array_agg
+ - [ ] avg
+ - [ ] bit_and
+ - [ ] bit_or
+ - [ ] bit_xor
+ - [ ] bool_and
+ - [ ] bool_or
+ - [ ] collect_list
+ - [ ] collect_set
+ - [ ] corr
+ - [ ] count
+ - [ ] count_if
+ - [ ] count_min_sketch
+ - [ ] covar_pop
+ - [ ] covar_samp
+ - [ ] every
+ - [ ] first
+ - [ ] first_value
+ - [ ] grouping
+ - [ ] grouping_id
+ - [ ] histogram_numeric
+ - [ ] kurtosis
+ - [ ] last
+ - [ ] last_value
+ - [ ] max
+ - [ ] max_by
+ - [ ] mean
+ - [ ] median
+ - [ ] min
+ - [ ] min_by
+ - [ ] mode
+ - [ ] percentile
+ - [ ] percentile_approx
+ - [ ] regr_avgx
+ - [ ] regr_avgy
+ - [ ] regr_count
+ - [ ] regr_intercept
+ - [ ] regr_r2
+ - [ ] regr_slope
+ - [ ] regr_sxx
+ - [ ] regr_sxy
+ - [ ] regr_syy
+ - [ ] skewness
+ - [ ] some
+ - [ ] std
+ - [ ] stddev
+ - [ ] stddev_pop
+ - [ ] stddev_samp
+ - [ ] sum
+ - [ ] try_avg
+ - [ ] try_sum
+ - [ ] var_pop
+ - [ ] var_samp
+ - [ ] variance
+
+### array_funcs
+ - [ ] array
+ - [ ] array_append
+ - [ ] array_compact
+ - [ ] array_contains
+ - [ ] array_distinct
+ - [ ] array_except
+ - [ ] array_insert
+ - [ ] array_intersect
+ - [ ] array_join
+ - [ ] array_max
+ - [ ] array_min
+ - [ ] array_position
+ - [ ] array_remove
+ - [ ] array_repeat
+ - [ ] array_union
+ - [ ] arrays_overlap
+ - [ ] arrays_zip
+ - [ ] flatten
+ - [x] get
+ - [ ] sequence
+ - [ ] shuffle
+ - [ ] slice
+ - [ ] sort_array
+
+### bitwise_funcs
+ - [x] &
+ - [x] ^
+ - [ ] bit_count
+ - [ ] bit_get
+ - [ ] getbit
+ - [x] shiftright
+ - [ ] shiftrightunsigned
+ - [x] |
+ - [ ] ~
+
+### collection_funcs
+ - [ ] array_size
+ - [ ] cardinality
+ - [ ] concat
+ - [x] reverse
+ - [ ] size
+
+### conditional_funcs
+ - [x] coalesce
+ - [x] if
+ - [ ] ifnull
+ - [ ] nanvl
+ - [x] nullif
+ - [ ] nvl
+ - [x] nvl2
+ - [x] when
+
+### conversion_funcs
+ - [ ] bigint
+ - [ ] binary
+ - [ ] boolean
+ - [x] cast
+ - [ ] date
+ - [ ] decimal
+ - [ ] double
+ - [ ] float
+ - [ ] int
+ - [ ] smallint
+ - [ ] string
+ - [ ] timestamp
+ - [ ] tinyint
+
+### csv_funcs
+ - [ ] from_csv
+ - [ ] schema_of_csv
+ - [ ] to_csv
+
+### datetime_funcs
+ - [ ] add_months
+ - [ ] convert_timezone
+ - [x] curdate
+ - [x] current_date
+ - [ ] current_timestamp
+ - [x] current_timezone
+ - [ ] date_add
+ - [ ] date_diff
+ - [ ] date_format
+ - [ ] date_from_unix_date
+ - [x] date_part
+ - [ ] date_sub
+ - [ ] date_trunc
+ - [ ] dateadd
+ - [ ] datediff
+ - [x] datepart
+ - [ ] day
+ - [ ] dayofmonth
+ - [ ] dayofweek
+ - [ ] dayofyear
+ - [x] extract
+ - [ ] from_unixtime
+ - [ ] from_utc_timestamp
+ - [ ] hour
+ - [ ] last_day
+ - [ ] localtimestamp
+ - [ ] make_date
+ - [ ] make_dt_interval
+ - [ ] make_interval
+ - [ ] make_timestamp
+ - [ ] make_timestamp_ltz
+ - [ ] make_timestamp_ntz
+ - [ ] make_ym_interval
+ - [ ] minute
+ - [ ] month
+ - [ ] months_between
+ - [ ] next_day
+ - [ ] now
+ - [ ] quarter
+ - [ ] second
+ - [ ] timestamp_micros
+ - [ ] timestamp_millis
+ - [ ] timestamp_seconds
+ - [x] to_date
```

Comment:
With Spark 3.4, `to_date` with no format arg translates to `cast(expr as date)`, which we do not currently support (but will soon - there is PR pending) and Comet cannot run natively because `Unsupported cast from StringType to DateType`.

When a format arg is supplied, Comet cannot run natively because `gettimestamp is not supported`.

Do you know why this doc says that it is supported?

Reply:
I think the problem here when Spark evaluates function of literal it skips Comet... the test above tests the function of the column and Comet enabled. Thinking how to fix it